Gems/ScriptCanvas/Code/CMakeLists.txt

```diff
@@ -106,6 +108,20 @@ ly_add_target(
             ${SCRIPT_CANVAS_AUTOGEN_BUILD_DIR}
 )
 
+ly_add_target(
+    NAME ScriptCanvas.AutoGen STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        scriptcanvasgem_autogen_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include/ScriptCanvas/AutoGen
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+)
+
```

Comment:
I think you move the target definition for `ScriptCanvas.Autogen` above the `ScriptCanvas.Static` since it is used by `ScriptCanvas.Static`

Reply:
Another approach is to make this a SHARED library, that is marked as a build dependency of `ScriptCanvas.Static`, so that any library that links to `ScriptCanvas.Static` would link the `ScriptCanvas.AutoGen` dll statically.

That would ensure only one copy is in memory at a time and would be a good pilot for seeing how well statically linked dlls work.

Any thoughts @jeremyong-az ?

Reply:
Will move above, btw the purpose is to avoid other gem/project dependes on ScriptCanvas.Static directly as build dependency which has too much not required scriptcanvas code. 
As @lsemp3d recommended to rename this target as ScriptCanvas.Extensions, this target will have limited and scoped code only.